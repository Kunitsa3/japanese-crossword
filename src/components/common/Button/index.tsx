import { cn } from '@/lib/utils';
import { FC } from 'react';

interface ButtonProps
  extends React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
  variant?: 'primary' | 'secondary';
}

export const Button: FC<ButtonProps> = ({ children, className, variant = 'primary', ...props }) => {
  return (
    <button
      {...props}
      className={cn(
        'px-4 py-1.5 text-base font-medium uppercase text-cyan-700 hover:text-cyan-600',
        className,
        variant === 'primary' && 'rounded border border-cyan-700'
      )}
    >
      {children}
    </button>
  );
};
