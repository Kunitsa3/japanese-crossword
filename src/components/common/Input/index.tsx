import { InputHTMLAttributes } from 'react';
import { FieldErrors, FieldValues, Path, UseFormRegister } from 'react-hook-form';

interface InputProps<T extends FieldValues> extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  name: Path<T>;
  errors: FieldErrors<T>;
  register: UseFormRegister<T>;
}

export const Input = <T extends FieldValues>({
  label,
  name,
  placeholder,
  errors,
  register,
  ...inputProps
}: InputProps<T>) => (
  <div>
    <label className="mb-2 mt-6 block text-sm font-medium text-cyan-700 dark:text-white">{label}</label>

    <input
      {...inputProps}
      {...register(name)}
      placeholder={placeholder}
      className="focus:cyan-700 block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 outline-none focus:border-cyan-700"
    />
    {errors[name] && <p className="text-red-700">{errors[name]?.message?.toString()}</p>}
  </div>
);
