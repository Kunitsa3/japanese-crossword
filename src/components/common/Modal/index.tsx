import { FC, ReactNode } from 'react';
import { createPortal } from 'react-dom';

interface ModalProps {
  children: ReactNode;
  setModalClosed: () => void;
}

export const Modal: FC<ModalProps> = ({ children, setModalClosed }) =>
  createPortal(
    <div className="fixed	inset-x-0 inset-y-0 top-20 z-10">
      <div className="h-full bg-neutral-200 opacity-90" onClick={setModalClosed} />
      <div className="fixed left-1/2 top-36 z-10 w-1/2 -translate-x-1/2 ">
        <div className="relative m-auto max-w-lg bg-white px-7 py-10 shadow">
          <div
            className="absolute right-8 top-2 cursor-pointer text-3xl text-cyan-700 hover:text-cyan-600"
            onClick={setModalClosed}
          >
            Ã—
          </div>
          {children}
        </div>
      </div>
    </div>,
    document.body
  );

export default Modal;
