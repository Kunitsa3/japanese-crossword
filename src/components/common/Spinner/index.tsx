import { cn } from '@/lib/utils';
import { FC } from 'react';

interface SpinnerProps {
  className?: string;
}

export const Spinner: FC<SpinnerProps> = ({ className }) => (
  <div className="flex h-full items-center justify-center">
    <div
      className={cn('h-10 w-10 animate-spin rounded-full border-4 border-dashed border-cyan-700 text-base', className)}
    />
  </div>
);

export default Spinner;
