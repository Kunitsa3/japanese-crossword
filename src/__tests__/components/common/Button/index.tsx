import { Button } from '@/components/common/Button';
import { cleanup, render } from '@testing-library/react';

afterEach(cleanup);

test('test button with default variant', () => {
  const { asFragment } = render(<Button />);

  expect(asFragment()).toMatchSnapshot();
});

test('test button with secondary variant', () => {
  const { asFragment } = render(<Button variant="secondary" />);

  expect(asFragment()).toMatchSnapshot();
});

test('test button with child text', () => {
  const buttonText = 'Text Button';
  const { getByText } = render(<Button variant="secondary">{buttonText}</Button>);

  expect(getByText(buttonText)).toHaveTextContent(buttonText);
});
