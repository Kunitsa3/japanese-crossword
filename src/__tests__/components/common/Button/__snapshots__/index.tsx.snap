// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button with default variant 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-1.5 text-base font-medium uppercase text-cyan-700 hover:text-cyan-600 rounded border border-cyan-700"
  />
</DocumentFragment>
`;

exports[`test button with secondary variant 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-1.5 text-base font-medium uppercase text-cyan-700 hover:text-cyan-600"
  />
</DocumentFragment>
`;
